src/Linux/Avalonia.LinuxFramebuffer/Output/DrmOutput.cs

```diff
@@ -203,47 +214,66 @@ public RenderSession(DrmOutput parent, IDisposable clearContext)
 
                 public void Dispose()
                 {
+                    using var clearContext = _clearContext;
+
                     _parent._deferredContext.GlInterface.Flush();
                     _parent._eglSurface.SwapBuffers();
-                    
+
                     var nextBo = gbm_surface_lock_front_buffer(_parent._gbmTargetSurface);
                     if (nextBo == IntPtr.Zero)
                     {
                         // Not sure what else can be done
                         Console.WriteLine("gbm_surface_lock_front_buffer failed");
+                        return;
                     }
-                    else
-                    {
 
-                        var fb = _parent.GetFbIdForBo(nextBo);
-                        bool waitingForFlip = true;
+                    DoPageFlipOrModeSet(nextBo);
 
-                        drmModePageFlip(_parent._card.Fd, _parent._crtcId, fb, DrmModePageFlip.Event, null);
+                    gbm_surface_release_buffer(_parent._gbmTargetSurface, _parent._currentBo);
+                    _parent._currentBo = nextBo;
+                }
 
-                        DrmEventPageFlipHandlerDelegate flipCb =
-                            (int fd, uint sequence, uint tv_sec, uint tv_usec, void* user_data) =>
-                            {
-                                waitingForFlip = false;
-                            };
-                        var cbHandle = GCHandle.Alloc(flipCb);
-                        var ctx = new DrmEventContext
+                private void DoPageFlipOrModeSet(IntPtr nextBo)
+                {
+                    if (_parent._deferedModesetActivator is not null)
```

Comment:
I think this feature is not supported on our SDK. Please update

Reply:
Downgrade code *
Our CI build doesn't work with latest SDK yet.